website/case-studies/implementation-of-flint-modules.md

```diff
@@ -0,0 +1,407 @@
+
+A module is a self-contained set of operations that determine the state of, or change in, variables across a specified period of time for a single Simulation Unit in direct response to event notifications from the FLINT core system (Unit Controller). For example, the empirical forest growth module includes all the operations required to simulate biomass accumulation.
+
+Modules are all written to work with the [`develop`](https://github.com/moja-global/FLINT/tree/develop) branch of the **FLINT** and any actively maintained branches will work with the most recent version (and are broken with old enough builds of the FLINT). The FLINT Application Programming Interface (API) has evolved over time, so some modules might be out of date. None of these modules require a custom fork of FLINT.
+
+## Background
+
+It is useful to think about what a module manifest might look like. **Manifests** are meta-data used by the package managers to describe the compatibility (FLINT versions that are used), dependencies (necessary inputs) and the configuration requirements (for example whether the module uses daily, monthly or annual time steps).
+
+This case study describes the core contribution of each module.
+
+## Index
+
+1.  [Prerequisites](#prerequisites)
+2.  [How do modules work?](#how-do-modules-work)
+3.  [Empirical forest growth module](#empirical-forest-growth-module)
+4.  [Hybrid forest growth module](#hybrid-forest-growth-module)
+5.  [WOFOST crop growth module](#wofost-crop-growth-module)
+6.  [Turnover module](#turnover-module)
+7.  [Decomposition module](#decomposition-module)
+8.  [RothC soil carbon module](#rothc-soil-carbon-module)
+9.  [Chapman Richards module](#chapman-richards-module)
+10. [GCBM module](#gcbm-module)
+11. [Frequently asked questions](#frequently-asked-questions)
+
+## Prerequisites
+
+These are the prerequistes you need to know before learning about **Modules**. This section presumes that you are already familiar with **FLINT**. If not, see [the FLINT repository](https://github.com/moja-global/FLINT) to better understand **FLINT** before continuing.
+
+## How do modules work?
+
+Each module reads (or is provided with – see Unit Controller) information about the current state variables and the data required to update the state variables, such as climate data or information about disturbances to simulate. Each module then performs the required calculations and returns information about the updates to apply to each of the state variables and C pools. 
+
+For state variables such as age or height it is possible for modules to return the updated value but for all carbon pools and other fluxes the module returns an array (sparse matrix) of the proposed operations. This array will include the information about the source pool, the sink pool, and the amount (Mg ha-1 per time step) of the flux. Module-specific metadata regarding units and time step size are also required. This information is all returned to the Unit Controller.
```

Comment:
This is quite technical. I suggest some feedback from @chicken-biryani and @sarahhaggarty about accessibility. Maybe there needs to be some more explanation to help users under what is being described. More examples might help.